build_engine_test_images/scripts/generate_images.sh

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env bash
+
+arch=$1
+docker_id=$2
+docker_password=$3
+docker_repo=$4
+
+# Get information
+apt install jq -y
+docker pull longhornio/longhorn-engine:master-head
+IMAGE="longhornio/longhorn-engine:master-head"
+
+version=`docker run $IMAGE longhorn version --client-only`
+echo Image version output: $version
+
+export CLIAPIVersion=`echo $version|jq -r ".clientVersion.cliAPIVersion"`
+export CLIAPIMinVersion=`echo $version|jq -r ".clientVersion.cliAPIMinVersion"`
+export ControllerAPIVersion=`echo $version|jq -r ".clientVersion.controllerAPIVersion"`
+export ControllerAPIMinVersion=`echo $version|jq -r ".clientVersion.controllerAPIMinVersion"`
+export DataFormatVersion=`echo $version|jq -r ".clientVersion.dataFormatVersion"`
+export DataFormatMinVersion=`echo $version|jq -r ".clientVersion.dataFormatMinVersion"`
+
+upgrade_image="${docker_repo}:upgrade-test.$CLIAPIVersion-$CLIAPIMinVersion"\
```

Comment:
Is it possible to reorganize and put similar code snippets or same functionality code into functions to get a clear view of this script?

Reply:
I reduced some unnecessary echo and variable declaration